GameTimeManagerクラスの実装

Moto0116 opened the pull request:
**追加機能**
- GameTimeManagerクラスの実装
- GameTimeEventクラスの実装
- GameTimeEventListenerクラスの実装
- GameDataManagerにゲーム時間の追加
- GameSceneでEventManagerクラスとGameDataManagerクラスを生成するように

issue #256

Moto0116 commented:
作業完了したのでレビューお願いします。

Keitaro0408 commented on src/KingdomCraft/KingdomCraft/Main/SceneManager/SceneBase/GameScene/EventManager/Event/GameTimeEvent/GameTimeEvent.h:
```diff
@@ -0,0 +1,82 @@
+		NONE,
+		NORMAL_OF_TIME,		// 時間の通常化
+		ACCELERATE_OF_TIME,	// 時間の加速
+		DECELERATE_OF_TIME,	// 時間の原則
```
漢字間違い。
あとOFは無くても良いと思う

Moto0116 replied:
指摘を受けた部分を修正しました確認お願いします。

Keitaro0408 commented on src/KingdomCraft/KingdomCraft/Main/SceneManager/SceneBase/GameScene/ObjectManager/GamaObjectManager/GameTimeManager/GameTimeManager.h:
```diff
@@ -0,0 +1,138 @@
+		THREE,			// 2倍速
+		FOUR,			// 4倍速
+		FIVE,			// 6倍速
+		PAUSE_STATE,	// 停止状態
```
~_SPEEDとかのほうがわかりやすいと思う

Moto0116 replied:
指摘を受けた部分を修正しました。
定数部分は~_SPEEDの形に変更したのと、enum名もSPEED_STATEに変更しました。
確認お願いします。

Keitaro0408 commented:
確認しました、コミットまとめお願いします

motolz committed:
[issue #256] GameTimeManagerクラスの実装

- GameTimeManagerクラスの実装
- GameTimeEventクラスの実装
- GameTimeEventListenerクラスの実装
- GameDataManagerにゲーム時間の追加
- GameSceneでEventManagerクラスとGameDataManagerクラスを生成するように

Moto0116 commented:
コミットログまとめときました。

Keitaro0408 commented:
合流しておきます
